src/macaron/output_reporter/results.py

```diff
@@ -203,9 +203,11 @@ def add_dep_record(self, dep_record: Record) -> None:
         dep_record : Record
             The record of the dependency.
         """
-        self.root_record.dependencies.append(dep_record)
-        if dep_record.context:
-            self.record_mapping[dep_record.context.remote_path] = dep_record
+        # Do not add a dependency if it's a duplicate.
+        if dep_record.record_id not in self.record_mapping:
```

Comment:
If we don't put the `dep_record` instance into `record_mapping` here, won't we lose the information for this duplicated dependency and it won't be printed out in the final result :thinking: ? I think we still add the dep_record to `self.root_record.dependencies` and we only need to change how the key of the `self.record_mapping` dictionary (to using record_id instead).

Reply:
I'm already using the `record_id` as the key to store records into `self.record_mapping`. But by adding the inclusion check the records in `self.root_record.dependencies` will also be consistent with `self.record_mapping`, which is necessary to avoid keeping two records with same `record_id`s in the dependencies (duplicates). Note that if two dependencies have different IDs but same remote paths, we still will add them both to dependencies and in the final report they will be shown as duplicates.